assets/js/base/context/cart-checkout/payment-methods/payment-method-data-context.js

```diff
@@ -337,23 +337,35 @@ export const PaymentMethodDataProvider = ( { children } ) => {
 	// Set active (selected) payment method as needed.
 	useEffect( () => {
 		const paymentMethodKeys = Object.keys( paymentData.paymentMethods );
+		const allPaymentMethodKeys = [
+			...paymentMethodKeys,
+			...Object.keys( paymentData.expressPaymentMethods ),
+		];
 		if ( ! paymentMethodsInitialized || ! paymentMethodKeys.length ) {
 			return;
 		}
 
 		setActive( ( currentActivePaymentMethod ) => {
 			// If there's no active payment method, or the active payment method has
 			// been removed (e.g. COD vs shipping methods), set one as active.
+			// Note: It's possible that the active payment method might be an
+			// express payment method. So registered express payment methods are
+			// included in the check here.
 			if (
 				! currentActivePaymentMethod ||
-				! paymentMethodKeys.includes( currentActivePaymentMethod )
+				! allPaymentMethodKeys.includes( currentActivePaymentMethod )
 			) {
```

Comment:
The offending condition was `! paymentMethodKeys.includes( currentActivePaymentMethod )`  for the following reasons:

- This effect callback is primarily for ensuring that there is an active payment method set when none is set.
- The value for `activePaymentMethod` could represent either a 'regular' payment method or an express payment method.
- The condition was only checking registered _payment methods_ for whether the key existed (not express payment methods).

So the fix was to ensure express payment methods are included in this conditional check as well.